superset/assets/src/dashboard/components/Dashboard.jsx

```diff
@@ -221,6 +241,38 @@ class Dashboard extends React.PureComponent {
     }
   }
 
+  openModal(filterKey) {
+    const slice_id = getModalSliceIDFor(this.props.dashboardState.publishSubscriberMap, filterKey);
+    if (slice_id) {
+      // open modal with chart in loading state
+      this.updateModalProps(slice_id, "loading", true)
+    }
+  }
+
+  updateModalProps(slice_id, status, showModal) {
+    if (showModal) {
+      this.modalChartStatus = status;
+      this.showModal = showModal
+      this.modalChart = this.props.charts[slice_id];
+      this.modalChart.chartStatus = status;
+      this.modalTitle = this.props.slices[slice_id].slice_name;
+      this.modalDatasource = this.props.datasources[this.modalChart.formData.datasource];
+    } else {
```

Comment:
else condition is not needed. We are setting this.showModal in both case.

Reply:
done